src/reducers/customLayers.ts

```diff
@@ -0,0 +1,73 @@
+import { GeoJSON } from 'geojson'
+import { ADD_CUSTOM_LAYER, REMOVE_CUSTOM_LAYER, TOGGLE_CUSTOM_LAYER, SET_CUSTOM_COLOR } from '../actions/customLayers'
+import { LOAD_CONFIGURATION, RESET_CONFIGURATION } from '../actions/saves'
+
+export interface CustomLayer {
+  filename: string
+  geoJSON: GeoJSON
+  zIndex: number
+  displayed: boolean
+  color: string
+}
+
+// Named colors used so screen readers can read them in ColorPicker
+export const customLayerColors = ['mediumSeaGreen', 'salmon', 'cornflowerBlue', 'orchid', 'hotPink']
+
+const defaultLayer: CustomLayer = {
+  filename: '',
+  geoJSON: { type: 'Polygon', coordinates: [] },
+  zIndex: 2,
+  displayed: true,
+  color: customLayerColors[0],
+}
+
+const getLeastUsedColor = (state: CustomLayer[]) => {
+  const colorsCount = new Array(customLayerColors.length).fill(0)
+  state.forEach((layer: CustomLayer) => {
```

Comment:
`layer: CustomLayer` is unnecessary, since TS should already know from `state: CustomLayer[]` what the argument to `forEach` will be.